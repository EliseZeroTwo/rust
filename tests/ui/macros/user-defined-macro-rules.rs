// check-fail

macro_rules! macro_rules { () => {} } //~ ERROR: user-defined macro may not be named `macro_rules`

macro_rules! {} //~ ERROR: expected identifier, found `{`
