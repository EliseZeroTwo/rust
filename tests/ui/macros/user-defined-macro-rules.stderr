error: user-defined macro may not be named `macro_rules`
  --> $DIR/user-defined-macro-rules.rs:3:14
   |
LL | macro_rules! macro_rules { () => {} }
   |              ^^^^^^^^^^^

error: expected identifier, found `{`
  --> $DIR/user-defined-macro-rules.rs:5:14
   |
LL | macro_rules! {}
   |              ^ expected identifier
   |
   = note: maybe you have forgotten to define a name for this `macro_rules!`

error: aborting due to 2 previous errors

